baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +36,41 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def fetch_data_from_url(url: str) -> Optional[str]:
+        data = None
+        try:
+            # Fetch the data from the url as bytes.
+            data = requests.get(url).content
+        except requests.exceptions.RequestException as e:
+            logger.exception(e)
+            return None
+        return data
+
+    @staticmethod
+    def get_data_to_write(json_data: dict, data: bytes) -> Optional[bytes]:
+        # Check if we have a JSON in this special format:
+        # data = {
+        #    "live_data_watcher_load_type": str
+        #    "data": str
+        #    "md5_hashed_data": str
+        # }
+        # If the load type is 'http', this format is an indication that we support
+        # downloading the contents of files uploaded to S3, GCS, etc when provided
+        # with an accessible URL.
+        if json_data.get("live_data_watcher_load_type") == "http":
+            # Only write the data if we actually managed to fetch its contents.
+            url = json_data.get("data")
+            if url is None:
+                logger.exception("No url found in zk source JSON node.")
```

Comment:
this is not in a try..exception so there's no exception in the context, thus we should not be using `logger.exception`?